### PR TITLE
A very large number of Science QoL tweaks and mapping changes.

### PR DESCRIPTION
**Pull request by Timberpoes**
## About The Pull Request

Reworked how Science access works.

ACCESS_TOX used to control access to R&D, except when it didn't and ACCESS_ROBOTICS was used as well, and used to control access to Toxins, except when it didn't and ACCESS_TOX_STORAGE was used instead.

This meant that some Science staff that probably shouldn't have access to Toxins (like Geneticists) also couldn't access R&D. I could have added ACCESS_GENETICS to R&D too, but I decided to pull apart Science access levels.

ACCESS_TOX is gone, replaced by ACCESS_TOXINS. ACCESS_TOX_STORAGE is gone, replaced by ACCESS_TOXINS_STORAGE. A new access level has been added, ACCESS_RND.

Now staff can be granted R&D Lab access without ALSO being given (or not given depending on the map) Toxins access. Robotics access has been removed from all R&D doors and windoors in favour of R&D access, meaning people can now be given Robotics lab access without ALSO being given (or not depending on the map) R&D access. Toxins Storage is now its own access level, incase you wanted to give someone access to the Toxins Lab without giving them free reign over a bunch of plasma tanks. Alternatively, you can now give them Toxins Storage access in all maps except Delta and they won't be able to use the tanks in to Toxins lab, but will still be able to access the discrete Toxins Storage area.

Access sweeps have been done across a number of science doors to accomodate these changes.

All science staff should find they have the same access to areas before and after the changes, except Geneticists who now have access to RnD.
All scientists have ACCESS_RND.
Scisec have ACCESS_RND.

Access isn't the only thing covered. Geneticists rejoice as Pubby and Delta have had genetics lab reworks. No longer do you have to punt eachother out of the way to access monkey cube boxes and no longer is there an imbalance of DNA Consoles and Scanners. All labs now contain three DNA Consoles/Scanner pairs and the two most cramped labs have been rearranged for a bit more space.

Meanwhile, Xenobiology has also had a host of tweaks.

All Xenobiology Labs now come equipped with Winter Coats and Boots incase someone breaches your freezer.
Xenobiology Lab's roundstart Winter Coats are now the science departmental variants.
All Xenobiology containment areas now have a Connector with an Air Injector.

Pubby now spawns 3 grey slimes instead of 2 on round start.

Meta's Xenobio Lab Freezer has been unhooked from its air distro loop, so it will no longer freeze Xenobiologists to death.

Delta's Xenobiology lab now requires Xenobiology access to enter, instead of ordinary Research department access.

But wait, there's more! Toxins mains rejoice! Pubby now has an intercom and Delta's intercom has been moved closer to the tachyon-doppler research array. Unfortunately, the RD's Prototype Hardsuit has been moved into the RD Office on all maps where it was in the Toxins Mass Driver room.

Box:
R&D Windoor now only has ACCESS_RND
Genetics Maint Door has had ACCESS_SCIENCE removed
Toxins Storage only has ACCESS_TOXINS_STORAGE
Toxins Lab only has ACCESS_TOXINS
Toxins Launch Room Access only has ACCESS_TOXINS
Toxins Mass Driver Windoor only has ACCESS_TOXINS
Adds Air Alarm to Genetics Lab
Adds Space Cleaner to Genetics Lab
Adds Box of Body Bags to Genetics Lab
Adds Bottle of Mutadone Pills to Genetics Lab
Monkey Cube Boxes separated in Genetics Lab
Removes Prescription Glasses from Genetics Lab
Xenobiology Containment Connector Port replace with Injector.

Pubby:
R&D Door only has ACCESS_RND and ACCESS_RD (As it also blocks access to the RD's office)
R&D Windoor only has ACCESS_RND
Toxins Storage only has ACCESS_TOXINS_STORAGE
Toxins Lab only has ACCESS_TOXINS
Toxins Mass Driver Windoor only has ACCESS_TOXINS
Nanite Lab no longer requires ACCESS_XENOBIOLOGY and instead requires ACCESS_RND
Maint Door leading to the Xenobio/Genetics checkpoint now requires ACCESS_GENETICS or ACCESS_XENOBIOLOGY instead of only ACCESS_MAINT_TUNNELS
Intercom added to Toxins Mass Driver Room
Genetics Lab re-designed. Fully equipped with 3 DNA Consoles and 3 DNA Scanners and the utmost in office-based tushie comfort.
Cautery added to Robotics Lab.
Prototype Hardsuit moved to RD Office.
3rd Grey Slime spawn added to Xenobio slime pens.
Xenobiology Lab now has 8 slime pens.
image
image

Meta: 
Prototype Hardsuit moved to RD Office.
Toxins access requirements moved to ACCESS_TOXINS
Toxins Storage access requirement moved to ACCESS_TOXINS_STORAGE
Removed Geneticist spawn point from the monkey pen.
Moved Genetics Lab security camera so that it can see the entire Genetics Lab including the Air Alarm on the North wall.
The Xenobiology Lab Freezer is now no longer sharing pipe space with Air Supply Pipes, freezing the air in Xenobiology's isolated air setup. As a result, stealing the now-moved RD's Prototype Hardsuit is no longer necessary to conduct Xenobiology.

Delta:
Removed Mannitol pills from Genetics.
Added Mutadone pills to Genetics.
Adds Space Cleaner to Genetics Lab.
Genetics Lab re-designed. Fully equipped with 3 DNA Consoles and 3 DNA Scanners.
Genetics Lab now requires ACCESS_GENETICS instead of also allowing ACCESS_RESEARCH. Anyone wanting to go to the Research Testing Range can go via Maintenance.
Genetics Lab's Medical Laptop now accepts Genetics access to view Medical Records. Fun!
Toxins Lab now requires ACCESS_TOXINS
Toxins Lab Secure Storage now requires ACCESS_TOXINS_STORAGE
Xenobiology Lab now requires ACCESS_XENOBIOLOGY instead of ACCESS_RESEARCH
Xenobiology Lab now has 8 slime pens.
Xenobiology Maintenance Doors now require ACCESS_XENOBIOLOGY instead of ACCESS_RESEARCH
Xenobiology Containment now has a Connector and Injector pair.
Moved Toxins Mass Driver room intercom to be within range of the tachyon-doppler research array.
image
image

## Why It's Good For The Game

Reworks two terrible Genetics labs.

Rethinks how Science access works, splitting Toxins from the R&D Lab.

Fixes a number of nagging issues with Xenobiology, Genetics, Toxins.

Adds a Cautery to the Pubby Robotics Lab. Welding Tool ghetto operations scared me.

Other quality of life changes.

## Changelog
:cl:
tweak: Delta's Genetics Lab has been redesigned.
tweak: Pubby's Genetics Lab has been redesigned.
tweak: Science ID Card access levels have been redesigned. R&D now has its own dedicated access level with Toxins and Toxins Storage having their own dedicated access levels.
tweak: Science doors across all maps have had their access levels tweaked, especially around maintenance access and R&D access.
fix: Geneticists now have R&D Access across all maps.
tweak: Xenobiology's Winter Coats now have a new set of pockets that can hold all the same equipment as a lab coat.
tweak: All Xenobiology secure containment areas there were missing them, now have Connector/Injector pairs for injecting gasses directly into secure containment.
tweak: Pubby's Xenobiology lab now has three slime spawns.
tweak: Pubby's Xenobiology lab has been expanded with addtional containment pens.
fix: Meta's Xenobiology freezer no longer shares a pipe with Xenobio's air distribution pipes. Scientists should no longer freeze to death and space suits with internals are no longer necessary.
tweak: Delta's Xenobiology lab now requires Xenobiology access to enter.
tweak: Delta's Xenobiology lab has been expanded with additional containment pens.
fix: Pubby's Toxins Mass Driver room now has an intercom.
tweak: Delta's Toxins Mass Driver room's intercom has been moved closer to the Tachyon-Doppler Research Array.
tweak: Prototype Hardsuits have been moved to the RD's Office on from Toxins on Meta and Pubby.
tweak: All Genetics Labs across all station now come fully equipped with Body Bags, Space Cleaner and Mutadone.
fix: Box's Genetics Lab now has an air alarm.
fix: Meta's Genetics Lab's camera has been moved so the AI can see the entire lab, including the Air Alarm.
fix: Geneticists will no longer spawn in the monkey pen on Meta.
tweak: The Medical Laptop in Delta's Genetics lab can now be accessed by Geneticists.
fix: Pubby's Robotics Lab now has a Cautery!
tweak: Probably more minor quality of life changes I forgot to document as I made them.
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
❤️ 

One thing I wanted to ask, though: On Pubby, can someone in the upper leftmost geneticist position see and talk to someone who's in the rightmost geneticist position? It looks like there's a corner of a wall in the way.

**Comment by Timberpoes** [comment]
> 
> 
> ❤️
> 
> One thing I wanted to ask, though: On Pubby, can someone in the upper leftmost geneticist position see and talk to someone who's in the rightmost geneticist position? It looks like there's a corner of a wall in the way.
> 
> Also, is the rightmost geneticist position on Pubby able to hear the notifications from the two leftmost genetics consoles (that are made whenever they discover a mutation)?

I'm going to hazard a guess at no to the first and I don't know, I assume they can, I'll test it later to the second.

Considering the distance of the consoles on Meta, it's likely just the line of sight that could kill it. I'll have a look at redoing a bit of Pubby's Xenobio to open up a little more room. I'll probably steal the two tiles and move the Xeno disposals and glass tables up.

I'm also seeing potential to Meta up the Pubby Xenobio while I'm at it. There's potential to turn it into a 7 or 8 pen setup using R-Glass Windows between the pens and stealing one tile from maint. That could make up for Genetics stealing a bit of space.

**Comment by Timberpoes** [comment]
@ATH1909 Please see the updated setup for Pubby in the PR's opening post.

I decided to do a sweep on Xenobiology while I was at it. They have been given the Meta treatment and as such there are now 8 slime pens on Pubby and Delta.

**Comment by optimumtact** [comment]
map power creep in action

**Comment by Timberpoes** [comment]
> 
> 
> map power creep in action

Perhaps, but some of the maps are incredibly behind on changes. Xenobiology has 2 consoles, but 6 pens really isn't good enough for any decent level of sharing. It's on the point where it's easier for a single person to use 6 pens than it is for two people to use 3 pens. The only map I've seen dual Xenobiologists semi-regularly is Box with its 11 Xenobio pens allowing a 6/5 split.

Delta and Pubby's genetics labs are legitimately awful. They were just crammed into whatever space was convenient to facilitate the change from Genetics to a sci department. Pubby's monkey pen has more room than the geneticists do. Some genetics labs come with Mannitol. Geneticists don't even use mannitol. Was it once something that turned monkeys into humans rather than curing brain damage? Box doesn't have an air alarm in the genetics lab. There's no way to control the vents in it.

I see these as closer to quality of life changes. If I've gone too far and now certain maps are imbalanced or approaching imbalance because the genetics labs all come with 3 DNA Scanner/Computer pairs and Xenobiology labs come with 8 slime pens, throw me some specifics on what I pull back on.

**Comment by TheVekter** [comment]
I'm in the process of some pretty severe changes to Pubby's XB lab, so I'll hold off on that until yours gets merged.

Also, thank you for fixing access. It's always been a clusterfuck.

**Comment by TheVekter** [comment]
image
This t-junction doesn't go anywhere. Also this corner could probably use more detail somewhere, like a sink or something.

Also, good changes for genetics. Thank you for helping to clean up what I did.

**Comment by Timberpoes** [comment]
Good catch. I'll clean that up with an L pipe.

**Comment by Timberpoes** [comment]
New update includes a tweak to the "the Pubby Xenobio corner" for Vekter -> There's now two lockers there, with a shower replacing the old locker in the airlock-y room. It also has two disposals units; a normal one and one linking to the space outlet.

BONUS ROUND: The Delta bar disposals unit is now actually connected to the disposals pipe network.

File: _maps/map_files/BoxStation/BoxStation.dmm [hunk 1]
```diff
@@ -32007,8 +32004,18 @@
 	dir = 1
 	},
 /obj/structure/rack,
-/obj/item/clothing/suit/hooded/wintercoat,
 /obj/item/clothing/shoes/winterboots,
+/obj/item/clothing/shoes/winterboots,
+/obj/item/clothing/suit/hooded/wintercoat{
```
[the remark this answers is not in the document]
Comment (Timberpoes) [reply _maps/map_files/BoxStation/BoxStation.dmm:32009]: Thanks for the check. Your prompt reminded me that there are a number of departmental winter coats already in. I replaced the genetic ones with the science departmental ones instead.

**Pull request merged**